### PR TITLE
Graph plotting, Petri nets, one inconsistency measure for BPMN

### PR DESCRIPTION
**Pull request by beneditknopp**
New features:

- Functionalities for plotting (_org.tweetyproject.commons.Plotting_), in particular plotting graphs (_org.tweetyproject.graphs.util.GraphPlotter_)
- Petri nets and reachability graphs for Petri nets, also a class to build the reachability graph of a Petri net _(org.tweetyproject.logics.petri)_
- Stochastic walks on reachability graphs (_org.tweetyproject.logics.petri.syntax.reachability_graph_). For this purpose, reachability graphs feature a probability function
- An inconsistency measurement (Dead-End Inconsistency) for BPMN models that is based on a stochastic walk on the reachability graph of the corresponding Petri net. For this purpose, BPMN models can be converted to Petri nets (_org.tweetyproject.logics.bpm.parser.bpmn_to_petri_).
- An example _(org.tweetyproject.logics.bpm.examples.Example)_ that visualizes various BPMN models as well as their Petri nets and reachability graphs, and the Dead-End inconsistency value for a random probability function.

### CODE REVIEW

**Comment by anachronismus** [comment]
I added two more dependencies in my commit. bpm/Example.java runs fine for me, it draws the examples like you described :+1:

**Review by anachronismus — APPROVED** [review]
I added two more dependencies in my commit. bpm/Example.java runs fine for me, it draws the examples like you described :+1:

**Review by matthiasthimm — APPROVED** [review]

File: org-tweetyproject-logics-bpm/src/main/java/org/tweetyproject/logics/bpm/parser/bpmn_to_petri/PetriNetParser.java [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+	
+	/**
+	 * Create a transition associated with two nodes of the BPMN model,
+	 * e.g. between two activities
+	 * @param nodeA the first associated node
+	 * @param nodeB the second associated node
+	 */
+	private Transition createTransition(BpmnNode nodeA, BpmnNode nodeB) {
+		if(nodeB == null) {
+			System.out.println("nono");
```
Comment (beneditknopp) [inline org-tweetyproject-logics-bpm/src/main/java/org/tweetyproject/logics/bpm/parser/bpmn_to_petri/PetriNetParser.java:317]: tbd

File: org-tweetyproject-logics-bpm/src/main/java/org/tweetyproject/logics/bpm/analysis/DeadEndInconsistencyMeasure.java [hunk 2]
```diff
@@ -14,43 +14,74 @@
...
+import org.tweetyproject.logics.petri.syntax.reachability_graph.ReachabilityGraph;
+import org.tweetyproject.math.matrix.Matrix;
+import org.tweetyproject.math.probability.ProbabilityFunction;
 
 /**
- * The DeadEndInconsistency value of a BPMN model is the number of activities without any successor activities
+ * A BPMN model has an IndeterminateInconsistency value of 1 if there is an activity in this model 
+ * a) that is reachable from a start event via the model's sequence flow
+ * b) from where no end event can possibly be reached, 
+ * and 0 otherwise
```
Comment (beneditknopp) [inline org-tweetyproject-logics-bpm/src/main/java/org/tweetyproject/logics/bpm/analysis/DeadEndInconsistencyMeasure.java:48]: tbd, wrong definition

File: org-tweetyproject-logics-petri/src/main/java/org/tweetyproject/logics/petri/plotting/PetriNetPlotter.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	public PetriNetPlotter(Plotter plotter, PetriNet petriNet) {
+		super(plotter, petriNet);
+	}
+	
+	/**
+	 * @param node the node to retrieve the pretty name for
+	 * @return the label of the node that will be displayed in the graph
+	 */
+	@Override
+	protected String getPrettyName(PetriNetNode node) {
```
Comment (beneditknopp) [inline org-tweetyproject-logics-petri/src/main/java/org/tweetyproject/logics/petri/plotting/PetriNetPlotter.java:58]: tbd

File: org-tweetyproject-logics-petri/src/main/java/org/tweetyproject/logics/petri/syntax/PetriNet.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+		return false;
+	}
+
+	@Override
+	public boolean isWeightedGraph() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+	
+	public void fire(Transition transition) {
```
Comment (beneditknopp) [inline org-tweetyproject-logics-petri/src/main/java/org/tweetyproject/logics/petri/syntax/PetriNet.java:218]: tbd

**Pull request merged**